validphys2/src/validphys/n3fit_data.py

```diff
@@ -92,7 +92,9 @@ def tr_masks(data, replica_trvlseed):
         cuts = dataset.cuts
         ndata = len(cuts.load()) if cuts else dataset.commondata.ndata
         frac = dataset.frac
-        trmax = int(frac * ndata)
+        trmax = int(frac*ndata)
+        if ndata == 1:
+            trmax = int(np.random.rand() <= frac)
```

Comment:
Can we use math.ceil everywhere instead?

Reply:
If we do then the 1-point datasets will always go to training though.

Reply:
Is that a bad thing?

Reply:
Yes, in the sense that it is not what we say we do.